src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.Fixer.cs

```diff
@@ -26,11 +28,123 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
-        {                              
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            return Task.Run(() => context.RegisterCodeFix(new MyCodeAction(ct => Fix(context, ct)), context.Diagnostics.First()));
+        }
+
+        private static async Task<Document> Fix(CodeFixContext context, CancellationToken cancellationToken)
+        {
+            var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var generator = context.Document.Project?.LanguageServices?.GetService<SyntaxGenerator>();
+            var root = await context.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            var diagnostic = context.Diagnostics.First();
+            switch (diagnostic.Properties[OperatorOverloadsHaveNamedAlternatesAnalyzer.DiagnosticKindText])
+            {
+                case OperatorOverloadsHaveNamedAlternatesAnalyzer.AddAlternateText:
+                    var methodDeclaration = generator.GetDeclaration(node, DeclarationKind.Operator) ?? generator.GetDeclaration(node, DeclarationKind.ConversionOperator);
+                    var operatorOverloadSymbol = (IMethodSymbol)semanticModel.GetDeclaredSymbol(methodDeclaration, context.CancellationToken);
+                    var typeSymbol = (ITypeSymbol)operatorOverloadSymbol.ContainingSymbol;
+                    var typeDeclarationSyntax = typeSymbol.DeclaringSyntaxReferences.First().GetSyntax(context.CancellationToken);
+                    var typeDeclaration = generator.GetDeclaration(typeDeclarationSyntax, DeclarationKind.Class);
```

Comment:
Re `struct`: the `GetDeclaration()` method also works for structs and I've added tests to verify this.

Re `typeDeclarationSyntax`: for C# these two nodes are the same, but for VB they're different and that's why this extra step is needed.  I'll add a comment explaining this here.